cmd/snap-confine/snap-confine.c

```diff
@@ -292,6 +294,11 @@ static void enter_classic_execution_environment(const sc_invocation *inv, gid_t
 static void enter_non_classic_execution_environment(sc_invocation *inv, struct sc_apparmor *aa, uid_t real_uid,
                                                     gid_t real_gid, gid_t saved_gid);
 
+static const cap_value_t snap_confine_caps_extra_cgroup_v1[] = {
```

Comment:
```suggestion
/* when operating with cgroup v1 we need to termporarily switch uid,
 * see comments in enter_non_classic_execution_environment() 
 */
static const cap_value_t snap_confine_caps_extra_cgroup_v1[] = {
```

Reply:
maybe worth to mention that distro packaging may have chosen to drop setuid/setgid, rendering the package unusable for uses cases of container on cgroup v1 host, but it's a fair choice on their side anyway

Reply:
yes, please, let's add a comment here